presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushLimitThroughProject.java

```diff
@@ -32,7 +32,11 @@
     private static final Capture<ProjectNode> CHILD = newCapture();
```

Comment:
> Identity projection could be a pruning projection therefore
we shouldn't push limit through it because rule executions could
loop.

Why it could loop?

Reply:
`PruneLimitColumns and `PushLimitThroughProject` could loop. Former creates pruning projection and later pushes limit through it and so on, so on...

Reply:
Add this information to commit message and to comment.